changes/2557-PrettyWood.md

```diff
@@ -0,0 +1,9 @@
+Refactor the whole _pydantic_ `dataclass` decorator to really act like its standard lib equivalent.
+It hence keeps `__eq__`, `__hash__`, ... and makes comparison with its non-validated version possible.
+It also fixes usage of `frozen` dataclasses in fields and usage of `default_factory` in nested dataclasses.
+The support of `Config.extra` has been added.
+Finally, config customization directly via a `dict` is now possible.
+<br/><br/>
+**BREAKING CHANGES**
+- The `compiled` boolean (whether _pydantic_ is compiled with cython) has been moved from `main.py` to `version.py`
```

Comment:
does this really count as breaking? The import from main will still work.

Reply:
`from pydantic.main import compiled` will fail